readme.md
# My Portfolio

## Images
<div align="center">

![Home](image/home.png)
_Home section_

---
![Technologies](image/technologies.png)
_Technologies section_

---
![Projects](image/projects.png)
_Projects section_

---
![Lightroom](image/lightroom.png)  
_Lightroom score (average 96.5%)_
</div>

---
## TODO

- [x] Added my interests
- [ ] Add portfolio projects with images
- [ ] Add a button to show email using JS to prevent crawlers extraction

## License

GPL-3


changelog.md
# Changelog

## Version 1.13.0 (2022/07/10)

- Changed urls to a shortened

---

## Version 1.12.0 (2022/07/05)

- Added description meta tag
- Added alt to technologies images
- Added animation to swing up the technology title when focused
- Minified languages colors js file
- Improved performance, SEO and accessibility

---

## Version 1.11.0 (2022/07/05)

- Added print button to submodules
- Fixed blob size and position
- Centered `#home` text
- Automatically hide the "go to top" button when reaches in footer to be able to
  read
- Fixed projects wrong padding
- Changed the quote to an African proverb

---

## Version 1.10.1 (2022/07/05)

- Updated images in readme

---

## Version 1.10.0 (2022/07/04)

- Added WordPress to technologies

---

## Version 1.9.1 (2022/07/03)

- Fixed website URL

---

## Version 1.9.0 (2022/07/03)

- Added side shadows to projects box

---

## Version 1.8.0 (2022/07/03)

- Changed color pallette
- Generate sub colors by a base color
- moved all colors declarations to a partial

---

## Version 1.7.0 (2022/07/03)

- Added favicon

---

## Version 1.6.0 (2022/07/02)

- Added home icon to meta (used in submodules)

---

## Version 1.5.3 (2022/07/01)

- Changed border-radius
- Using `--border-radius` var
- Fixed `projects.sass` border-radius animation
- Added hover effect in `projects.sass` project
- Fixed projects element creation
- Updated the HTML projects cache

---

## Version 1.5.2 (2022/06/29)

- Fixed padding of Github fetched projects title

---

## Version 1.5.1 (2022/06/28)

- Added `rel="noopener noreferrer" target="_blank"` to all links
- Updated the projects "cache" to works without JS

---

## Version 1.5.0 (2022/06/28)

- Added "All glory to YAHUAH" footer text
- Fixed footer elements alignment

---

## Version 1.4.2 (2022/06/22)

- Checking if project has a license
- async download of projects

---

## Version 1.4.1 (2022/06/21)

- Moved blob to a partial

---

## Version 1.4.0 (2022/06/21)

- Added a quote

---

## Version 1.3.2 (2022/06/21)

- Fixed abstract
- Added more MVP config variables 

---

## Version 1.3.1 (2022/06/20)

- Changed duplicated font awesome pseudo elements to a mixin
- Minified some third party libs and minified generated style

---

## Version 1.3.0 (2022/06/16)

- Fixed separation title style

---

## Pre-version 1.3.0 (2022/06/15)

- Changed secondary color
- Added fish shell to technologies
- Fixed typo
- Added a title fix for printing

---

## Pre-version 1.3.0 (2022/06/14)

- Removed `devicon.min.css`
- Fixed media queries

---

## Version 1.2.0 (2022/06/12)

- Added Puppeteer

---

## Version 1.1.0 (2022/06/12)

- Added more interests
- Removed forks from repos count

---

## Version 1.0.3 (2022/06/11)

- Fixed Github API sort from `updated` to `pushed`

---

## Version 1.0.2 (2022/06/11)

- Minified the fetched data from Github to save space in local storage

---

## Version 1.0.1 (2022/06/11)

- Changed `updated_at` to `pushed_at`
- Fixed fetching
- removed debug alert
- Added rounded borders to Github icon in footer

---

## Version 1.0.0 (2022/06/11)

- Added local storage time based cache that leasts 10 minutes, if exceeds the
  time download again

---
## Version 0.6.0 (2022/06/11)

- Ordered projects by last updated
- Added fetched repos count
- Changed the projects subtitle
- Fixed the project description drawing
- Added repo creation date

---

## Version 0.5.1 (2022/06/11)

- Technology interest is a link that goes to `#technologies` section
- Removed static Github data intended to test

---

## Version 0.5.0 (2022/06/10)

- Added pulse animation to target project
- About me
  - Added interests
  - Added motivation
- Added Git to technologies
- Added no js projects view

---

## Version 0.4.0 (2022/06/10)

- Fixed github icon
- Added license
- Added todo

---

## Version 0.3.0 (2022/06/10)

- Added more technologies
- Added the github projects

---

## Version 0.2.0 (2022/06/09)

- Added technologies logo
- moved all third-party CSSs to `third` dir
- technologies section
  - Added more
  - Added description
  - Added url to tech website
  - Fixed link click to work well in mobile
  - Added bootstrap
  - Added Debian

---

## Version 0.1.2 (2022/06/06)

- Fixed blob s-ize in mobile
- Added a Y limit to blob
- Added info to footer
- fixed footer style

---

## Version 0.1.1 (2022/06/06)

- Fixed Go to top button
- Fixed my name fade colors
- fixed position of first section

---

## Version 0.1.0 (2022/04/30)

- copy from ECOleta
- 
Eu tenho mais [^1] pra dizer.

[^1]: diga aqui.	


public/cl/index.php
<?php
if ($_GET["passcode"] != "48a4957021b9cc0d70a4842d5261a7") {
  http_response_code(404);
  echo '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en"><head><title>404 &mdash; Not Found</title><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/><meta name="description" content="Sorry, page not found"/><style type="text/css">body{font-size:14px; color:#777777; font-family:arial; text-align:center} h1{font-size:180px; color:#99A7AF; margin: 70px 0 0 0} h2{color: #DE6C5D; font-family: arial; font-size: 20px; font-weight: bold; letter-spacing: -1px; margin: -3px 0 39px} p{width:320px; text-align:center; margin-left:auto;margin-right:auto; margin-top: 30px} div{width:320px; text-align:center; margin-left:auto;margin-right:auto} a:link{color: #34536A} a:visited{color: #34536A} a:active{color: #34536A} a:hover{color: #34536A} </style></head><body><p><a href="http://thiago.oxyoy.com/">thiago.oxyoy.com</a></p><h1>404</h1><h2>Page Not Found</h2><div>It seems that the page you were trying to reach does not exist anymore, or maybe it has just moved. You can start again from the <a href="http://thiago.oxyoy.com/">home</a> or go back to <a href="javascript:%20history.go(-1)">previous page</a>. </div></body></html>';
  die();
}
include "index.html";
?>


public/script/projects/projects.js
import getProjects from "./getProjects.js"
import languagesColor from "./languagesColor.js"
import projectElement from "./projectElement.js"

;(async (user) => {

  const projects = await getProjects(user)
  const container = document.querySelector("#projects .projects")

  container.innerHTML = ""

  let scrollTo = null
  let reposCount = 0

  projects.map(async (project) => {
    if (project.fork) return
    if (!project.language) return
    const el = projectElement(project, languagesColor[project.language].color)
    container.append(el)
    reposCount += 1
    if (window.location.hash.slice(1) == project.id) scrollTo = el
  })

  const updateStatus = document.querySelector(".updateStatus")
  updateStatus.innerHTML = `Fetched from Github <span class>${reposCount}</span> projects.`

  if (scrollTo) setTimeout(() => scrollTo.scrollIntoView(), 600)
})("thisago")


public/script/projects/getProjects.js
/**
 * Get the projects of user from localstorage or Github
 *
 * @param {string} user
 * @param {number} max Default 100
 * @param {number} page Default 1
 * @param {string} type Default "owner"
 * @param {string} sort Default "updated"
 * @param {boolean} forceFetch Force fetch from Github
 * @param {number} storageLife Time in seconds that data will be held in localstorage, if exeeds, then will fetch again; Default 10 minutes
 *
 * @returns {object} The projects of the user
 */
export default async function getProjects(
  user,
  max = 100,
  page = 1,
  type = "owner",
  sort = "pushed",
  forceFetch = false,
  storageLife = 10 * 60
) {
  let willFetch = forceFetch
  const currentData = localStorage.getItem("gh_repos"),
    limitDataTime = new Date(
      parseInt(localStorage.getItem("gh_repos_time")) + storageLife * 1000
    ),
    nowTime = new Date()
  if (!currentData) willFetch = true
  if (nowTime > limitDataTime) willFetch = true

  if (willFetch) {
    localStorage.setItem(
      "gh_repos",
      await fetchProjects(user, max, page, type, sort)
    )
    localStorage.setItem("gh_repos_time", Date.now())
    return await getProjects(user, max, page, type, sort, false, storageLife)
  } else return JSON.parse(currentData)
}

/**
 * Fetches the projects of user from Github
 *
 * @param {string} user
 * @param {number} max
 * @param {number} page
 * @param {string} type
 * @param {string} sort
 *
 * @returns {object} The projects of the user
 */
async function fetchProjects(
  user,
  max = 100,
  page = 1,
  type = "owner",
  sort = "pushed"
) {
  return JSON.stringify(
    await (
      await fetch(
        `https://api.github.com/users/${user}/repos?per_page=${max}&page=${page}&type=${type}&sort=${sort}`
      )
    ).json()
  )
}


public/script/projects/projectElement.js
/**
 * Generates the DOM element with project data
 *
 * @param {object} project
 * @param {string} langColor
 *
 * @returns {HTMLElement} Aside element
 */
export default function projectElement(project, langColor) {
  const id = `project_${project.id}`
  const aside = document.createElement("aside")
  aside.id = id
  aside.onclick = () => {
    location.hash = id
  }
  aside.innerHTML = `<h3><a rel="noopener noreferrer" target="_blank" href="${project.html_url}">${project.name}</a></h3>
<p>${project.description || ""}</p>
${
  project.homepage ? `
    <div class="homepage">
      <a rel="noopener noreferrer" target="_blank" href="${project.homepage}">${project.homepage}</a>
    </div>
  `.trim() : ``
}

${
  project.topics.length > 0 ? `
    <div class="tags">${
      "<span>" + project.topics.join("</span><span>") + "</span>"
    }</div>`.trim() : ""
}

<div class="meta">
  <div class="left">
    <span class="data language">
      <span class="languageColor" style="--lang-color: ${langColor}"></span>
      <span class="language">${project.language}</span>
    </span>
  </div>
  <div class="right">
    <span class="data stars">${project.stargazers_count}</span>
    <span class="data forks">${project.forks}</span>
    ${project.license ?
      "<span class=\"data license\">" + project.license.spdx_id + "</span>"
      : ""}
  </div>
</div>
<div class="dates">
  <small>Updated ${timeSince(new Date(project.pushed_at))} ago</small>
  <span></span>
  <small>Created ${timeSince(new Date(project.created_at))} ago</small>
</div>`.trim()
  return aside
}

function timeSince(date) {
  var seconds = Math.floor((new Date() - date) / 1000)
  var interval = Math.floor(seconds / 31536000)

  if (interval > 1) {
    return interval + " years"
  }
  interval = Math.floor(seconds / 2592000)
  if (interval > 1) {
    return interval + " months"
  }
  interval = Math.floor(seconds / 86400)
  if (interval > 1) {
    return interval + " days"
  }
  interval = Math.floor(seconds / 3600)
  if (interval > 1) {
    return interval + " hours"
  }
  interval = Math.floor(seconds / 60)
  if (interval >= 1) {
    return interval + " minutes"
  }
  return Math.floor(seconds) + " seconds"
}
